org.testeditor.web.dropwizard.xtext/src/test/java/org/testeditor/web/xtext/index/BuildCycleManagerTest.xtend

```diff
@@ -0,0 +1,197 @@
+package org.testeditor.web.xtext.index
+
+import org.eclipse.emf.common.util.URI
+import org.eclipse.xtext.build.BuildRequest
+import org.eclipse.xtext.build.IncrementalBuilder
+import org.eclipse.xtext.build.IndexState
+import org.eclipse.xtext.naming.QualifiedName
+import org.eclipse.xtext.resource.IResourceServiceProvider
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.eclipse.xtext.resource.impl.ChunkedResourceDescriptions
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData
+import org.eclipse.xtext.resource.persistence.SerializableEObjectDescription
+import org.eclipse.xtext.xbase.lib.Functions.Function1
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentCaptor
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.mockito.Spy
+import org.testeditor.web.dropwizard.xtext.XtextConfiguration
+import org.testeditor.web.dropwizard.xtext.validation.ValidationMarkerUpdater
+
+import static org.assertj.core.api.Assertions.assertThat
+import static org.mockito.ArgumentMatchers.*
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.verify
+import static org.mockito.Mockito.when
+
+class BuildCycleManagerTest {
+
+	@Spy var config = new XtextConfiguration => [
+		localRepoFileRoot = '/base/dir'
+		indexSearchPaths = #['/search/path']
+	]
+	@Mock IndexSearchPathProvider mockSearchPathProvider
+	@Mock ChangeDetector mockChangeDetector
+	@Mock ChangedResources mockChangedResources
+	@Mock XtextResourceSet mockResourceSet
+	@Mock ValidationMarkerUpdater mockValidationMarkerUpdater
+	@Mock IncrementalBuilder mockIncrementalBuilder
+	@Mock extension IResourceServiceProvider.Registry mockResourceServiceProviderRegistry
+	@Mock ChunkedResourceDescriptionsProvider indexProvider
+	@Mock ChunkedResourceDescriptions index
+	@InjectMocks extension BuildCycleManager unitUnderTest
```

Comment:
buildCycleManagerUnderTest (or something like that) is a better name in my eyes. 'unitUnderTest' is a bit too generic. Is it realy necessary to put this as extension into the test class? I'd rather not use extension here. Since there are a lot of mocks here, I would put a line break here, too.